source/templates.tex

```diff
@@ -3887,6 +3889,79 @@
 \end{codeblock}
 \end{example}
 
+\pnum
+If deduction against the other template succeeds for both transformed templates,
+constraints can be considered as follows:
+\begin{itemize}
+\item
+If their \grammarterm{template-parameter-list}{s}
+(possibly including \grammarterm{template-parameter}{s}
+invented for an abbreviated function template\iref{dcl.fct}) or
+function parameter lists differ in length,
+neither template is more specialized than the other.
+\item
+Otherwise:
+\begin{itemize}
+\item
+If exactly one of the templates was considered by overload resolution
+via a rewritten candidate with reversed order of parameters:
+\begin{itemize}
+\item
+If, for either template, some of the \grammarterm{template-parameter}{s}
+are not deducible from their function parameters,
+neither template is more specialized than the other.
+\item
+If there is either no reordering or more than one reordering
+of the associated \grammarterm{template-parameter-list}
+such that the corresponding \grammarterm{template-parameter}{s}
+of the \grammarterm{template-parameter-list}{s} are equivalent and
+such that the function parameters that positionally correspond
+between the two templates are of the same type,
+neither template is more specialized than the other.
```

Comment:
Ack!  Please add a separate editorial commit which breaks this horrendous sentence up via subbullets.

Reply:
Fixed. (third level of subbullets, yay!)